experimental/devices/sn3218/sn3218.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2018 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+package sn3218
+
+import (
+	"errors"
+
+	"periph.io/x/periph/conn/i2c"
+)
+
+const (
+	i2cAddress             = 0x54
+	cmdEnableOutput        = 0x00
+	cmdSetBrightnessValues = 0x01
+	cmdEnableLeds          = 0x13
+	cmdUpdate              = 0x16
+	cmdReset               = 0x17
+)
+
+// Dev is a handler to sn3218 controller.
+type Dev struct {
+	i2c        i2c.Dev
```

Comment:
you need to track if the device is in sleep mode or now, and transparently wake it up (Enable) on any call enabling any LED.

Reply:
Track the state is a good idea. But I disagree with wakeup on change. It should be possible to change the states for different channels and enable everything later at once.

Example use case (What I have in mind for my project)
```
        d.BrightnessAll(1)
        d.Switch(7, true) //Because a sensor mesaured something
        d.Switch(5, true) //Because a sensor measured something else
        d.WakeUp() //User pushed button to show the states
        time.Sleep(1000 * time.Millisecond)
        d.Sleep()  //Save power again
```

Reply:
Track sleepMode still TODO